### PR TITLE
Add session management feature to Boilerplate (#8019)

### LINKED ISSUE #8019: Boilerplate project template lacks session management feature

### PR DESCRIPTION
**Pull request by ysmoradi**
This closes #8019

### CODE REVIEW

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Client/Boilerplate.Client.Core/Components/Pages/Identity/Profile/DeleteAccountConfirmModal.razor.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ private async Task DeleteAccount()
     {
         await userController.Delete(CurrentCancellationToken);
 
-        await AuthenticationManager.SignOut();
+        await AuthenticationManager.SignOut(CurrentCancellationToken);
```
Comment (msangtarash) [inline src/Templates/Boilerplate/Bit.Boilerplate/src/Client/Boilerplate.Client.Core/Components/Pages/Identity/Profile/DeleteAccountConfirmModal.razor.cs:24]: No need to call UserController.SignOut inside AuthenticationManager

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Client/Boilerplate.Client.Core/Services/AuthenticationManager.cs [hunk 2]
```diff
@@ -30,15 +31,22 @@ public async Task<bool> SignIn(SignInRequestDto request, CancellationToken cance
...
-    public async Task SignOut()
+    public async Task SignOut(CancellationToken cancellationToken)
     {
-        await storageService.RemoveItem("access_token");
-        await storageService.RemoveItem("refresh_token");
-        if (AppPlatform.IsBlazorHybrid is false)
+        try
         {
-            await cookie.Remove("access_token");
+            await userController.SignOut(cancellationToken);
```
Comment (msangtarash) [inline src/Templates/Boilerplate/Bit.Boilerplate/src/Client/Boilerplate.Client.Core/Services/AuthenticationManager.cs:38]: Call UserController.SignOut only if refresh token exists

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Controllers/Identity/UserController.cs [hunk 3]
```diff
@@ -33,6 +35,64 @@ public async Task<UserDto> GetCurrentUser(CancellationToken cancellationToken)
...
+
+    [HttpPost]
+    public async Task SignOut(CancellationToken cancellationToken)
+    {
+        var userId = User.GetUserId();
+
+        var user = await userManager.FindByIdAsync(userId.ToString())
+            ?? throw new ResourceNotFoundException();
+
+        user.Sessions = user.Sessions.Where(s => s.SessionUniqueId != Guid.Parse(User.FindFirstValue("session-id")!)).ToList();
```
Comment (msangtarash) [inline src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Controllers/Identity/UserController.cs:59]: store `Guid.Parse(User.FindFirstValue("session-id")!` in variable

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Mappers/IdentityMapper.cs [hunk 4]
```diff
@@ -12,4 +13,23 @@ public static partial class IdentityMapper
...
+
+    public static UserSessionDto Map(this UserSession source, TimeSpan refreshTokenExpiresIn)
+    {
+        var dto = source.AutoMap();
+
+        dto.LastSeenOn = source.RenewedOn is null ||
+                         DateTimeOffset.UtcNow - source.RenewedOn < TimeSpan.FromMinutes(5)
+                         ? AppStrings.Online
+                         : DateTimeOffset.UtcNow - source.RenewedOn < TimeSpan.FromMinutes(15)
+                            ? AppStrings.Recently
```
Comment (msangtarash) [inline src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Mappers/IdentityMapper.cs:28]: Check if static usage of AppStrings works in all scenarios

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Controllers/Identity/IdentityController.cs [hunk 5]
```diff
@@ -211,24 +216,72 @@ public async Task<ActionResult<SignInResponseDto>> SignIn(SignInRequestDto reque
...
     public async Task<ActionResult<TokenResponseDto>> Refresh(RefreshRequestDto request)
     {
         var refreshTokenProtector = bearerTokenOptions.Get(IdentityConstants.BearerScheme).RefreshTokenProtector;
         var refreshTicket = refreshTokenProtector.Unprotect(request.RefreshToken);
 
-        if (refreshTicket?.Properties?.ExpiresUtc is not { } expiresUtc || DateTimeOffset.UtcNow >= expiresUtc ||
-                await signInManager.ValidateSecurityStampAsync(refreshTicket.Principal) is not User user)
+        UserSession? userSession = null;
+
+        if (refreshTicket?.Properties?.ExpiresUtc is not { } expiresUtc ||
```
Comment (msangtarash) [inline src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Controllers/Identity/IdentityController.cs:264]: if found user, but refresh token was expired, then remove session, if everything is fine, set sessions.RenewedOn, and in finally, save the user

**Review by msangtarash — CHANGES_REQUESTED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**